gcovr/configuration.py

```diff
@@ -50,6 +50,30 @@ def check_inputfile(value):
     return os.path.abspath(value)
 
 
+def check_output_file(value):
+    r"""
+    Check if the output file can be created.
+    """
+
+    if value is not None:
+        if os.path.isfile(value) and not os.access(value, os.W_OK):
+            raise ArgumentTypeError(
+                "File {value} exist but is not writable".format(value=value))
+
+        dirname = os.path.dirname(value)
+        if dirname == '':
+            dirname = '.'
```

Comment:
I thought hard about this line yesterday. This is correct for command line arguments. But this is not correct when the path is read from the config file.

This is not your fault, but the current code is buggy. We special-case filters in the argument parsing code to fix up relative paths, maybe we need something similar for files?

Should we fix it as part of this PR, or separately?

Reply:
This is not only for configuration files. Another solution here is to make the path absolute and return this path as it is done for input files.
Paths in config files should be treated relative to the config file and changed to an absolute path while reading the config file. I'll check this with this PR.

Reply:
Special treatment of check_input_file, check_output_file and OutputOrDefault. The new test uses the --html-css and --output options in the configfile.